### PR TITLE
test(search): refactor accessibility tests

### PR DESCRIPTION
**Pull request by ZhuoyuJin**
### Proposed behaviour

- Add `accessibility` Cypress tests for the `Search` component to use the new cypress-component-react framework for testing.
- Move component stories to the `search.stories.tsx`
- Refactor for `search.stories.mdx` file
- Move testing component to the `search-test.stories.tsx`
- Create the `search.stories.tsx`

### Current behaviour
Old approach for accessibility tests

### Checklist

- [x] Commits follow our style guide
- [ ] Related issues linked in commit messages if required
- [ ] Screenshots are included in the PR if useful
- [ ] All themes are supported if required
- [ ] Unit tests added or updated if required
- [x] Cypress automation tests added or updated if required
- [ ] Storybook added or updated if required
- [ ] Translations added or updated (including creating or amending translation keys table in storybook) if required
- [ ] Typescript `d.ts` file added or updated if required

#### QA

- [x] Tested in CodeSandbox/storybook
- [x] Add new Cypress test coverage if required
- [x] Carbon implementation matches Design System/designs
- [ ] UI Tests GitHub check reviewed if required

### Testing instructions

- [ ] Run `npx cypress open --component` to check if there is newly added test.file
- [ ] Check if the `search.test.js` file passed
- [ ] Run `npx cypress run --component` to check none of the other *.test.js files have regressed
- [ ] Run `npx cypress run --e2e` to check none of the feature files have regressed
- [ ] Run `npm run build-storybook` -> `npx sb extract` -> `npx http-server storybook-static -p 9001` and run `npx cypress run --e2e` and check that `Search` tests are not running twice.

### CODE REVIEW

**Review by DlgSHi — CHANGES_REQUESTED** [review]

File: src/components/search/search.stories.mdx [hunk 1]
```diff
@@ -211,89 +154,23 @@ For more information check our [Validations](?path=/docs/documentation-validatio
...
-  </Story>
+  <Story name="validations - string - component" story={stories.ValidationsStringComponent} />
 </Canvas>
 
 It is possible to use the `tooltipPosition` to override the default placement of tooltips rendered as part of this component.
 
 <Canvas>
   <Story
     name="validations - string - with tooltipPosition overriden - component"
-    parameters={{ chromatic: { disable: true } }}
```
[the remark this answers is not in the document]
Comment (ZhuoyuJin) [reply src/components/search/search.stories.mdx:244]: Thanks for the comment, this story is re-write in line 165

File: src/components/search/search.stories.mdx [hunk 2]
```diff
@@ -38,79 +37,45 @@ import Search from "carbon-react/lib/components/search";
...
-            name="test"
-            placeholder="Search..."
-            onChange={(e) => setValue(e.target.value)}
-            value={value}
-            searchButton
-          />
-        </div>
-      );
-    }}
-  </Story>
```
Comment (stephenogorman) [inline src/components/search/search.stories.mdx:69]: Has 'search.stories.tsx' file been created?

**Review by DlgSHi — APPROVED** [review]

File: src/components/search/search.test.js [hunk 3]
```diff
@@ -404,4 +392,154 @@ context("Test for Search component", () => {
...
+      }
+    );
+
+    it("should check accessibility for Search with tabIndex prop", () => {
+      CypressMountWithProviders(<SearchComponent tabIndex={-5} />);
+
+      cy.checkAccessibility();
+    });
+
+    // FE-4670
```
Comment (DipperTheDan) [inline src/components/search/search.test.js:480]: Is this comment meant to be here?

Comment (ZhuoyuJin) [reply src/components/search/search.test.js:480]: Thanks for reviewing it Dan, this comment is for the purpose of skipping the tests below as there are still accessibility violations in search. And this is the ticket in our backlog.

Comment (DipperTheDan) [reply src/components/search/search.test.js:480]: Didn't see the `.skip` 😆

**Review by DipperTheDan — APPROVED** [review]

**Pull request merged**